Combined updates from other pull requests

jeapostrophe opened the pull request.

stamourv committed:
Use `make-log-based-evalutator` from scribble/eval.

stamourv committed:
Use `make-log-based-evalutator` from scribble/eval.

AlexKnauth committed:
split pos, dir, linear, affine, and smooth into separate modules

AlexKnauth committed:
separate affine combinators into a separate module

mflatt committed:
adapt for high-resolution displays

Version 1.18 of "gui-lib" provides the `get-gl-client-size` method
that is needed to sync GL sizing with the canvas drawing scale.
Enable hires mode for Mac OS X when that support is available.

mflatt committed:
add x, y, and display-mode to `big-bang3d`

mflatt committed:
document x, y, and display-mode arguments

mflatt committed:
add a `gl-config` argument to `big-bang` and `pict3d-canvas%`

Also, add `pict3d-default-gl-config`.

mflatt committed:
make fullscreen-like mode auto-adjust on display change

AlexKnauth committed:
Add dependency on unstable-lib

for unstable/custom-write and unstable/sandbox

mflatt committed:
avoid time-based yield

mflatt committed:
avoid `unstable/custom-write`

mflatt committed:
big-band3d: add `#:cursor` option

mflatt committed:
request incremental GC

jeapostrophe committed:
Add export of shaders for external validation

jeapostrophe committed:
Merge pull request #1 from mflatt/master

Migrate

jeapostrophe committed:
Merge pull request #2 from stamourv/master

Use `make-log-based-evalutator` from scribble/eval.

jeapostrophe committed:
protect against release for incremental mode

AlexKnauth committed:
Merge pull request #3 from AlexKnauth/patch-1

Add dependency on unstable-lib

jeapostrophe committed:
Shader compilation test support and spec conformance

AlexKnauth committed:
Merge pull request #4 from AlexKnauth/pos2

split pos, dir, linear, affine, and smooth into separate modules

AlexKnauth committed:
update version dependency on base 6.3

because of racket/struct

jeapostrophe committed:
Merge pull request #5 from AlexKnauth/update-dep-version

update version dependency on base 6.3

mflatt committed:
big-bang3d: add `#:pause-state?`

mflatt committed:
avoid show & hide for GL context on OS X and Windows

Until `racket/gui` provides a better way to ensure that a GL
context is initialized, take platform-specific steps to
ensure that the invisible GL context is initialized
through its frame. In particular, showing and hiding the frame
 no longer works on OS X, because hiding disables the context.

AlexKnauth committed:
import cpointer? and ctype? using unsafe (#7)

to avoid using any-wrap/c on the inputs to the predicates

mflatt committed:
"Mac OS X" -> "Mac OS"

jeapostrophe committed:
Cleanup bitmap code

jeapostrophe committed:
Fixing bitmaps on OS X

jeapostrophe committed:
Shrink public pict3d interface

Commit:
(gl-extensions) needs to be called within with-gl-context
Clean up trailing whitespaces in viewer.rkt

jeapostrophe committed:
Merge pull request #14 from bluewww/master

(gl-extensions) needs to be called within with-gl-context in viewer.rkt

bb010g commented:
I think a hard fork is the best choice from here, given how @ntoronto was last active 30 June 2015.

Commit:
Re-enable texture-loading in utils.rkt. Fixes planet.rkt, bitmaptest.rkt and vbotest.rkt.

jeapostrophe committed:
Merge pull request #16 from bluewww/enable-texture-loading

Re-enable texture-loading in utils.rkt.

sorawee committed:
Add the license field to info.rkt

I use LGPL-3.0-only here. LGPL-3.0-or-later is another possibility,
but I opt for the conservative one. Let me know if this should be
changed.

As a side-effect, this should trigger a rebuild, which hopefully will
fix the current build failure at
https://pkg-build.racket-lang.org/server/built/fail/pict3d.txt

jeapostrophe committed:
Merge pull request #27 from sorawee/add-license

Add the license field to info.rkt

rfindler committed:
remove dependency on unstable-lib (#29)

samth committed:
Use flasin/flacos to avoid complex numbers.

Related to https://github.com/racket/typed-racket/pull/1428

jeapostrophe committed:
Merge pull request #33 from samth/master
